controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java

```diff
@@ -2631,6 +2637,24 @@ private StreamSegmentRecord newSegmentRecord(int epoch, int segmentNumber, long
 
     abstract CompletableFuture<Map<UUID, ActiveTxnRecord>> getTxnInEpoch(int epoch, OperationContext context);
 
+    /**
+     * Method to get active txns according to their status.
+     *
+     * @param context operational context
+     * @param status  transaction status
+     * @return list of transaction ids having given status
+     */
+    abstract CompletableFuture<List<UUID>> listTransactionsInActiveStates(OperationContext context, TxnStatus status);
```

Comment:
nit: could we call this `listActiveTransactions`?

Reply:
rename method as suggested.